lib/omnibus/builder.rb

```diff
@@ -563,6 +564,29 @@ def delete(path, options = {})
     expose :delete
 
     #
+    # Strip symbols from the given file or directory on the system. This method uses
+    # find and passes the matched files to strip through xargs, ignoring errors.
+    # So one may pass in a specific file/directory or a glob of files.
+    #
+    # @param [String] path
+    #   the path of the file(s) to strip
+    #
+    # @return (see #command)
+    #
+    def strip(path)
+      regexp_ends = ".*(" + IGNORED_ENDINGS.map { |e| e.gsub(/\./, '\.') }.join("|") + ")$"
+      regexp_patterns = IGNORED_PATTERNS.map { |e| ".*" + e.gsub(/\//, '\/') + ".*" }.join("|")
+      regexp = regexp_ends + "|" + regexp_patterns
+
+      # Do not actually care if strip runs on non-strippable file, as its a no-op.  Hence the `|| true` appended.
+      # Do want to avoid stripping files unneccessarily so as not to slow down build process.
+      find_command = "find #{path}/ -type f -regextype posix-extended ! -regex '#{regexp}' | xargs strip || true"
```

Comment:
The single quotes here will probably fail on windows because of the funky way we shellout here https://github.com/chef/omnibus/blob/master/lib/omnibus/util.rb#L92  It's a `bash -c 'your command'` that runs through cmd and they both escape stuff differently.

On the other hand, if this works on *nix systems, I'm ok with just coming back later and fighting with the windows compat issue.  We'll probably end up using your previous ruby-based algorithm on windows anyway and eating the built-time cost on windows because that's usually how this stuff works out.